src/Utilities/StdConstexprHelpers.hpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/* These functions are constexpr reimplementations of some standard library
+ * functions, designed to work with cpp17::array. Therefore, these functions
+ * are in the cpp17 namespace.
+ */
+
+#pragma once
+
+#include "Utilities/Gsl.hpp"
+
+namespace cpp17 {
```

Comment:
Please put these in the namespaces and files we discussed on Slack. They should also be added to the `Utilities` group in Doxygen.  They should also have tests that test they work in a constexpr context.

Reply:
Sure. I’m unclear precisely which namespaces you want from the slack discussion. Can you please clarify here for each function?

Reply:
Never mind, I figured out which are cpp20 :)